src/CMakeLists.txt

```diff
@@ -814,7 +814,8 @@ if(WITH_LIBCEPHFS)
       OUTPUT_NAME cephfs
       VERSION 2.0.0
       SOVERSION 2)
-    if(NOT APPLE)
+    if(NOT APPLE AND NOT
+        (WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL Clang))
```

Comment:
I noticed those issues while preparing an MSI installer. The CI job uses static linking while the MSI build script uses dynamic linking, which is why they only came up now.

I don't expect any further changes at this point.

Reply:
> I noticed those issues while preparing an MSI installer. The CI job uses static linking while the MSI build script uses dynamic linking, which is why they only came up now.

This is important context that would have been nice to include as a comment before pushing.

> 
> I don't expect any further changes at this point.

So for now Windows builds are going with `ENABLED_SHARED` set to `OFF` but the intention is to flip to `ON` when/if Boost is fixed?

Reply:
> This is important context that would have been nice to include as a comment before pushing.

Right, sorry for not mentioning it. 

> So for now Windows builds are going with `ENABLED_SHARED` set to `OFF` but the intention is to flip to `ON` when/if Boost is fixed?

Windows CI builds use ``ENABLE_SHARED=OFF``, mostly due to some tests that were failing to build otherwise: https://github.com/ceph/ceph/blob/bd86572d5a139d1213161f30fe1834f9974b1922/win32_build.sh#L77-L79

I had a full build with ``ENABLE_SHARED=ON`` a few days ago, so I think we managed to address the issue in the meantime. We can switch the default build to ``ENABLE_SHARED=ON`` as a subsequent PR, or I can just add another commit here.

Reply:
I'd lean towards a separate PR, just to avoid further changes here that aren't strictly related to the switch to LLVM.